util/design/gen-otp-mmap.py

```diff
@@ -51,46 +51,51 @@ def main():
-        else:
-            random.seed()
-            new_seed = random.getrandbits(64)
-            if config.setdefault('seed', new_seed) == new_seed:
-                log.warning(
-                    'No seed specified, setting to {}.'.format(new_seed))
-
-        otp_mmap = OtpMemMap(config)
-
-        with open(PARTITIONS_TABLE_FILE, 'w') as outfile:
-            outfile.write(TABLE_HEADER_COMMENT +
-                          otp_mmap.create_partitions_table())
-
-        with open(DIGESTS_TABLE_FILE, 'w') as outfile:
-            outfile.write(TABLE_HEADER_COMMENT +
-                          otp_mmap.create_digests_table())
-
-        with open(MMAP_TABLE_FILE, 'w') as outfile:
-            outfile.write(TABLE_HEADER_COMMENT + otp_mmap.create_mmap_table())
-
-        # render all templates
-        for template in TEMPLATES:
-            with open(template, 'r') as tplfile:
-                tpl = Template(tplfile.read())
-                with open(
-                        Path(template).parent.joinpath(Path(template).stem),
-                        'w') as outfile:
-                    outfile.write(tpl.render(otp_mmap=otp_mmap))
+    try:
+        args = parser.parse_args()
+
+        with open(MMAP_DEFINITION_FILE, 'r') as infile:
+            config = hjson.load(infile)
+
+            # If specified, override the seed for random netlist constant computation.
+            if args.seed:
+                log.warning('Commandline override of seed with {}.'.format(
+                    args.seed))
+                config['seed'] = args.seed
+            # Otherwise, we either take it from the .hjson if present, or
+            # randomly generate a new seed if not.
+            else:
+                random.seed()
+                new_seed = random.getrandbits(64)
+                if config.setdefault('seed', new_seed) == new_seed:
+                    log.warning(
+                        'No seed specified, setting to {}.'.format(new_seed))
+
+            otp_mmap = OtpMemMap(config)
+
+            with open(PARTITIONS_TABLE_FILE, 'w') as outfile:
+                outfile.write(TABLE_HEADER_COMMENT +
+                              otp_mmap.create_partitions_table())
+
+            with open(DIGESTS_TABLE_FILE, 'w') as outfile:
+                outfile.write(TABLE_HEADER_COMMENT +
+                              otp_mmap.create_digests_table())
+
+            with open(MMAP_TABLE_FILE, 'w') as outfile:
+                outfile.write(TABLE_HEADER_COMMENT + otp_mmap.create_mmap_table())
+
+            # render all templates
+            for template in TEMPLATES:
+                with open(template, 'r') as tplfile:
+                    tpl = Template(tplfile.read())
+                    with open(
+                            Path(template).parent.joinpath(Path(template).stem),
+                            'w') as outfile:
+                        outfile.write(tpl.render(otp_mmap=otp_mmap))
+
+    except Exception as err:
```

Comment:
I'd consider just catching `RuntimeError` here and/or moving the try/catch block inwards. The problem is that `Exception` includes all the crazy things that e.g. Mako might generate when rendering a template.

Reply:
Ok got it, done!